src/NHibernate/Linq/DefaultQueryProvider.cs

```diff
@@ -80,24 +101,6 @@ protected virtual NhLinqExpression PrepareQuery(Expression expression, out IQuer
 			return nhLinqExpression;
 		}
 
-		private static readonly MethodInfo Future = ReflectHelper.GetMethodDefinition<IQuery>(q => q.Future<object>());
-		private static readonly MethodInfo FutureValue = ReflectHelper.GetMethodDefinition<IQuery>(q => q.FutureValue<object>());
-
-		protected virtual object ExecuteFutureQuery(NhLinqExpression nhLinqExpression, IQuery query, NhLinqExpression nhQuery)
-		{
-			var method = (nhLinqExpression.ReturnType == NhLinqExpressionReturnType.Sequence ? Future : FutureValue)
```

Comment:
This condition was guaranteeing the `ToFutureValue` without selector case could only ever have a `IEnumerable<T>` result, causing the later commented `if` below to be moot. `ReturnType` is always sequence when the expression evaluate to `IQueryable`, see [here](https://github.com/nhibernate/nhibernate-core/blob/master/src/NHibernate/Linq/NhLinqExpression.cs#L56). Expression as only been [`PreTransform`](https://github.com/nhibernate/nhibernate-core/blob/master/src/NHibernate/Linq/NhRelinqQueryParser.cs#L56) at that point, and that does not change its type.